src/Orleans/Lifecycle/ILifecycleObserver.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System.Threading.Tasks;
+
+namespace Orleans
+{
+    /// <summary>
+    /// Lifecycle observer used to handle start and stop notification.
+    /// </summary>
+    public interface ILifecycleObserver
+    {
+        /// <summary>
+        /// Handle start notifications
+        /// </summary>
+        /// <returns></returns>
+        Task OnStart();
```

Comment:
This is handled by the extension functions (See LifecycleExtensions), which allow for a user to pass in individual functions rather than a full observer.  For instance one could do something like:

Task MyConfig() {...}
Task MyInit() {...}
Task MyFinit() {...}
Task MyRun() {...}

lifcycle.Subscribe(Stages.Config, MyConfig):
lifcycle.Subscribe(Stages.Init, MyInit, MyFinit):
lifcycle.Subscribe(Stages.Run, MyRun):

Registering each callback at the appropriate stage.